src/main/java/com/puppycrawl/tools/checkstyle/FileStatefulModule.java

```diff
@@ -0,0 +1,48 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2017 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * This annotation means that the module contains file-related context and therefore
+ * cannot be used from the separate threads at the same time.
+ * This annotation should be used when a module holds a thread-unsafe state.
```

Comment:
module is: all that is referenced in XML as class - AuditLister, Filters, FileFilters, Checks, Composite Checks(TreeWalker).
Is it really for all of them ?
If for all - please do not use "the Checks" in further doc at all. If not - please use right terms.

Reply:
@romani For FileSet and AbstractCheck modules. Is it correct to name them as "checks"?

Should we also rename annotations? They are named [Stateless / FileStateful / GlobalStateful]**Module**

Reply:
> Is it correct to name them as "checks"?

yes.

> Should we also rename annotations?

yes, if you target only Checks.

Reply:
@soon , is it fixed ?

Reply:
Fixed